### DISCUSSION TITLE
How to get number of plugins installed?

### DISCUSSION THREAD

**Discussion by Maritsu** [wbthomason/packer.nvim#1209] [Q&A] [answered]
Like the title says, I'm looking for a way to get the number/amount of currently installed plugins. Does anyone know whether it's possible to do this? If so, how could I achieve this?
Thanks in advance

**Comment by yutkat** [comment, marked as answer]
```lua
local function get_table_size(t)
	local count = 0
	for _, __ in pairs(t) do
		count = count + 1
	end
	return count
end

get_table_size(_G.packer_plugins)
```

**Reply by Maritsu** [reply]
`_G.packer_plugins` seems to not be defined from what I can see while trying this answer.
![image](https://user-images.githubusercontent.com/66056347/218318868-2f0029d8-5c92-402e-b37d-dd4b5ac12a28.png)

**Reply by yutkat** [reply]
It should be defined in the compiled packer configuration file.

```lua
util.join_paths(vim.fn.stdpath('config'), 'plugin', 'packer_compiled.lua') -- ~/.config/nvim/plugin/packer_compiled.lua'
```

**Reply by Maritsu** [reply]
Sorry for the dumb question, but what does that code do? I've only really touched Lua via Neovim configuration

**Reply by yutkat** [reply]
`_G.packer_plugins` is only available after `packer_compiled.lua` is loaded so cannot be used till after plugins have been loaded.

You might want to write it down at the very end of your configuration.

**Reply by Maritsu** [reply]
Oh, aight!  
just asking while I remember - is there a way to wait till `packer_compiled.lua` is loaded?

**Reply by Maritsu** [reply]
> It should be defined in the compiled packer configuration file.
> 
> ```lua
> util.join_paths(vim.fn.stdpath('config'), 'plugin', 'packer_compiled.lua') -- ~/.config/nvim/plugin/packer_compiled.lua'
> ```

Also, what's `util` in this situation? Is this just an excerpt from `packer_compiled.lua` or should I paste this at the end of my config?

**Reply by yutkat** [reply]
> is there a way to wait till packer_compiled.lua is loaded?

After require('packer').startup, I think it's loaded.

I was looking at the source code and thought that it might be better to use this function. This way it's not timing dependent.

https://github.com/wbthomason/packer.nvim/blob/1d0cf98a561f7fd654c970c49f917d74fafe1530/lua/packer/plugin_utils.lua#LL101C10-L101C10


> what's util in this situation? 

util is this one.

```lua
util = require 'packer.util'
```

It's also in the example setup.

https://github.com/wbthomason/packer.nvim#custom-initialization

**Reply by Maritsu** [reply]
In that case, what function should I use to get the plugin count so that it's not "timing dependent"?

(Also, getting the plugin count is done after `require("packer").startup()` in my case. Hell, it's done right at the end of the config)

**Reply by yutkat** [reply]
```lua
local function get_table_size(t)
  local count = 0
  for _, __ in pairs(t) do
    count = count + 1
  end
  return count
end
local opt, start = require('packer.plugin_utils').list_installed_plugins()
local plugin_count = get_table_size(opt) + get_table_size(start)
print(plugin_count)
```

**Reply by yutkat** [reply]
Using `_G.packer_plugins` in `init.lua` was not correct because it is loaded after the `init.lua` file is loaded. 
`require('packer.plugin_utils').list_installed_plugins()` is a better way to do this.

**Reply by Maritsu** [reply]
It finally works, thank you very much!!
